src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -478,8 +479,18 @@ private ObjectTypeDiscriminatorImplementation[] CreateDiscriminatorImplementatio
                 .ToHashSet();
         }
 
+        private HashSet<string> GetParentPropertyDeclarationNames()
+        {
+            return EnumerateHierarchy()
+                .Skip(1)
+                .SelectMany(type => type.Properties)
+                .Select(p => p.Declaration.Name)
+                .ToHashSet();
+        }
+
         protected override IEnumerable<ObjectTypeProperty> BuildProperties()
         {
+            var propertiesFromSpec = GetParentPropertyDeclarationNames();
```

Comment:
well in the `ModelTypeProvider` variant, we are using the `ISymbol` to indicate whether the member has been handled. Here we are using their names, which I believe should work fine - still I have a concern when there is name collisions, whether our system could handle that edge case (symbol definitely will not have this issue).
It is fine that we keep it as is here right now, because in a near future, we will not have this class any more, but if there is any issue, we might keep in mind that this implementation potenially has this issue.